src/lib.rs

```diff
@@ -308,12 +278,10 @@ pub fn encrypt_and_generate_constraints<F: Field>(
         computed_ciphertext.extend_from_slice(&ciphertext_chunk);
     }
 
-    for (i, byte) in ciphertext.iter().enumerate() {
-        byte.enforce_equal(
-            computed_ciphertext
-                .get(i)
-                .to_anyhow("Error getting ciphertext byte")?,
-        )?;
+    // finally, we insert the computed ciphertext as a public input of the circuit
+    for byte in &computed_ciphertext {
+        let value = byte.value().map_err(|e| anyhow!("{}", e))?;
```

Comment:
Here you can use `e.to_string()` if I'm not mistaken.
```suggestion
        let value = byte.value().map_err(|e| anyhow!(e.to_string()))?;
```